libs/execution/src/lib/types/io-types/filesystem-inmemory.ts

```diff
@@ -4,94 +4,96 @@
+    return null;
   }
 
-  getFile(filePath: string): BinaryFile | null {
-    const processedParts = this.processPath(filePath);
+  putFile(path: string, file: BinaryFile): FileSystem | null {
+    const processedParts = this.processPath(path);
+    let currentDir = this.rootDirectory;
 
-    // If we have a minimum valid path
-    if (processedParts.length > 0) {
-      let currentFileSystemIndex = this.fileSystemIndex;
-
-      // Get the name of the requested file
-      const fileName = processedParts[processedParts.length - 1] as string;
+    // Loop until we reach the last part of processedParts
+    for (let i = 0; i < processedParts.length - 1; i++) {
+      const part = processedParts[i];
 
-      // If we have a directory, we traverse it, if not (processedParts == 1) skip the travesal
-      for (let i = 0; i < processedParts.length - 1; i++) {
-        const childFileSystem = currentFileSystemIndex.get(
-          processedParts[i] as string,
-        ) as InMemoryFileSystem | undefined;
+      // If part exists
+      if (part != null) {
+        let childNode = currentDir.getChildNode(part);
 
-        // If we dont find current path-part, stop methodcall
-        if (!childFileSystem) {
+        // If dir exists --> traverse
+        if (childNode && childNode instanceof FileSystemDirectory) {
+          currentDir = childNode;
+        } else if (childNode && childNode instanceof FileSystemFile) {
           return null;
+
+          // If dir NOT exists --> create new
+        } else {
+          childNode = new FileSystemDirectory(part);
+          currentDir.add(childNode);
+          currentDir = childNode as FileSystemDirectory;
         }
-        currentFileSystemIndex = childFileSystem.fileSystemIndex;
       }
+    }
 
-      // If we are in the correct directory
-      const file = currentFileSystemIndex.get(fileName) as
-        | BinaryFile
-        | undefined;
-      return file === undefined ? null : file;
+    // If Nodename is not already there --> put file to dir
+    const child = currentDir.getChildNode(processedParts.pop() ?? '');
+    if (child instanceof FileSystemDirectory || child === undefined) {
+      currentDir.add(file);
     }
     return null;
   }
 
-  putFile(filePath: string, file: BinaryFile): FileSystem {
-    const processedParts = this.processPath(filePath);
-    let currentFileSystemIndex = this.fileSystemIndex;
-
-    // If we need to traverse
-    for (let i = 0; i < processedParts.length - 1; i++) {
-      const part: string = processedParts[i] as string;
-
-      // Check, if directory already exists, if not create it
-      let childFileSystemIndex = currentFileSystemIndex.get(part) as
-        | InMemoryFileSystem
-        | undefined;
-      if (!childFileSystemIndex) {
-        childFileSystemIndex = new InMemoryFileSystem();
-        currentFileSystemIndex.set(part, childFileSystemIndex);
+  private processPath(path: string): string[] {
+    const parts = path
+      .toLowerCase()
```

Comment:
Why do you use `toLowerCase` here? That way, users cannot refer to files / folders that have capital letters in their names.

Reply:
@rhazn  once raised the requirement, to handle paths case insensitive, thats why i am using toLowerCase() (see https://github.com/jvalue/jayvee/pull/126#discussion_r1098412123)

Reply:
Hmm I thought that is how unix does it but apparently I was wrong, we can change to care for capitalization, yeah.

Reply:
okay, changed to case sensitive pathing